platforms/ts/wabt/WabtNativeHost.ts

```diff
@@ -0,0 +1,103 @@
+import * as typenames from "./WabtNativeHost-types.d";
+import wabt from 'wabt';
+
+// Parses a WebAssembly text format source to a module.
+export function parseWat(filename: string, buffer: string, options: typenames.WasmFeatures): typenames.Promise<typenames.WasmModule, string> {
+	let fn = (fulfill: (x: any) => void, reject: (x: any) => void) =>
+		wabt().then(w => {
+			const m = w.parseWat(filename, buffer, options);
+			if (m) {
+				const module : typenames.WasmModule = {
+					name: "WasmModule",
+					_id : -1,
+					validate: () => m.validate(),
```

Comment:
The functions that can throw should be catched in this place, and we should return a bool or a string to map error handling back to flow.

Reply:
The 'validate' function is wrapped with try-catch and exceptions are caught and exposed as non-empty strings with messages.